ZipBuilder/Sources/FirebaseReleaser/Tags.swift

```diff
@@ -36,12 +38,43 @@ enum Tags {
         fatalError("Non-Firebase pod \(pod.name) is missing a version")
       }
       let tag = pod.name.replacingOccurrences(of: "Google", with: "") + "-" + version
-      createTag(gitRoot: gitRoot, tag: tag)
+      createTag(gitRoot: gitRoot, tag: tag, deleteExistingTags: deleteExistingTags)
     }
   }
 
-  private static func createTag(gitRoot: URL, tag: String) {
+  static func updateTags(gitRoot: URL) {
+    createTags(gitRoot: gitRoot, deleteExistingTags: true)
+  }
+
+  private static func createTag(gitRoot: URL, tag: String, deleteExistingTags: Bool) {
+    if deleteExistingTags {
+      verifyTagsAreSafeToDelete()
+      Shell.executeCommand("git tag --delete \(tag)", workingDir: gitRoot)
+      Shell.executeCommand("git push --delete origin \(tag)", workingDir: gitRoot)
+    }
     Shell.executeCommand("git tag \(tag)", workingDir: gitRoot)
     Shell.executeCommand("git push origin \(tag)", workingDir: gitRoot)
   }
+
+  private static func verifyTagsAreSafeToDelete() {
+    var homeDirURL: URL
```

Comment:
Should we be extra safe and run a `pod update` here first to be sure that the local repo is up to date? I imagine it's a very very edge case but maybe something worth doing.

Reply:
Good catch! See #6774